### PR TITLE
py-nbmake: add v1.3.0; py-nbclient: add v0.6.6, 0.5.13; py-traitlets: add v5.3.0

### PR DESCRIPTION
**Pull request by adamjstewart**
Successfully installs on macOS 12.4 (Apple M1 Pro) with Python 3.9.13 and Apple Clang 13.1.6.

Depends on #31512

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-traitlets/package.py [hunk 1]
```diff
@@ -25,7 +26,8 @@ class PyTraitlets(PythonPackage):
 
     depends_on('python@3.7:', when='@5:', type=('build', 'run'))
     depends_on('python@2.7:2.8,3.3:', type=('build', 'run'))
-    depends_on('py-setuptools@40.8:', type='build')
+    depends_on('py-hatchling@0.25:', when='@5.2.1.post0:', type='build')
+    depends_on('py-setuptools@40.8:', when='@:5.2.1.a', type='build')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-traitlets/package.py:30]: Is this a typo on the version in the `when` clause (i.e., is `.a` deliberate)?

**Review by tldahlgren — COMMENTED** [review]
Confirmed all sha256.  Just have a question about one of the `when` clauses.

File: var/spack/repos/builtin/packages/py-traitlets/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-traitlets/package.py:30]: The `.a` is deliberate. No 5.2.1.a release exists, but `@:5.2.1` includes `5.2.1.post0` so we need to append a `.a` to cap the range before then.

**Review by tldahlgren — DISMISSED** [review]

**Comment by tldahlgren** [comment]
Looks like there are some missing dependencies.

**Comment by tldahlgren** [comment]
`py-traillets` appears to need `py-hatchlings` ~(for windows)~.

**Comment by adamjstewart** [comment]
Yes, hence:

> Depends on https://github.com/spack/spack/pull/31512

**Comment by tldahlgren** [comment]
Apologies, missed the `Draft` and making my way LIFO through the list.

**Comment by adamjstewart** [comment]
No worries, added the draft after you reviewed. I'm also going to add a couple newer versions.

**Review by tldahlgren — COMMENTED** [review]
LGTM

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**